### PR TITLE
fixed issue #297

### PR DESCRIPTION
**Pull request by brain-sick**
### When you change things like this in brackets please remember to delete brackets after you change it.

297# This pull request is related to issue(s) 

the player was one able to move through on of the door

# The problem(s) I want to solve or The facilitie(s) I want to improve is/are,
the player went through the tile the tile where key is placed without any gate, i fixed this and also let the player reach the treasure after it had the key though the only only gate being through the cricket ground title

###i edited the file tiles.js, i added a bool variable to check if key was found or not and then allowed entrance to the treasure 

### Links

http://rawgit.com/brain-sick/labyrinth/master/index.html

### CODE REVIEW

**Comment by niccokunzmann** [comment]
#297

File: js/tiles.js [hunk 1]
```diff
@@ -226,6 +226,7 @@ const OpenDoors = {
 
 // door is used in /js/levels.js ignoring the error
 // ignore JSHintBear
+var keyFound=false; //boolean variable to check if key is found or not
```
Comment (niccokunzmann) [inline js/tiles.js]: Could you use the inventory for that? @YashKumarVerma can guide you with this.

The problem with global variables is: If we may add a second player in the future, we can not use them any more. So the state needs to be in the player.
I think, most of what you have written there is correct and you can use the inventory to store and ask if the key is there.

File: js/tiles.js [hunk 2]
```diff
@@ -461,6 +473,7 @@ const door = {
         visit: function() {
             alertInfo("You got the Key for the Treasure !", "Explore More ! Let's see what can find ...");
             // add item to inventory
+            keyFound=true;
             player.inventory.add(['Key', 'key.png']);
```
Comment (niccokunzmann) [inline js/tiles.js]: see here for the inventory.

Comment (brain-sick) [reply js/tiles.js]: ok. i will try using inventory

**Comment by brain-sick** [comment]
the new clone i did alredy  had this issue solved

**Pull request closed without merging**